Experiences/Experiences/View Controllers/ImageSelectionViewController.swift

```diff
@@ -0,0 +1,167 @@
+import MapKit
+
+class ImageSelectionViewController: ShiftableViewController {
+    
+    var mapViewController: MapViewController?
+    var userLocation: CLLocationCoordinate2D?
+    private let context = CIContext()
+    private let exposureAdjustFilter = CIFilter.exposureAdjust()
+    private var selectedImage: UIImage? {
+        didSet {
+            guard let selectedImage = selectedImage else { return }
+            
+            var scaledSize = imageView.bounds.size
+            let scale = UIScreen.main.scale
+            
+            scaledSize = CGSize(width: scaledSize.width * scale, height: scaledSize.height * scale)
+            
+            let scaledUIImage = selectedImage.imageByScaling(toSize: scaledSize)
+            guard let scaledCGImage = scaledUIImage?.cgImage else { return }
+            
+            scaledImage = CIImage(cgImage: scaledCGImage)
+        }
+    }
+    
+    private var scaledImage: CIImage? {
+        didSet {
+            updateImage()
+        }
+    }
+    
+    @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var chooseImageButton: UIButton!
+    @IBOutlet weak var titleTextField: UITextField!
+    @IBOutlet weak var imageHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var recordAudioButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        recordAudioButton.isEnabled = false
+        titleTextField.delegate = self
+    }
+    
+    @IBAction func selectImage(_ sender: Any) {
+        
+        let authorizationStatus = PHPhotoLibrary.authorizationStatus()
+        
+        switch authorizationStatus {
+        case .authorized:
+            presentImagePickerController()
+        case .notDetermined:
+            
+            PHPhotoLibrary.requestAuthorization { (status) in
+                guard status == .authorized else {
+                    NSLog("User did not authorize access to the photo library")
+                    self.presentInformationalAlertController(title: "Error", message: "In order to access the photo library, you must allow this application access to it.")
+                    return
+                }
+                self.presentImagePickerController()
+            }
+            
+        case .denied:
+            self.presentInformationalAlertController(title: "Error", message: "In order to access the photo library, you must allow this application access to it.")
+        case .restricted:
+            self.presentInformationalAlertController(title: "Error", message: "Unable to access the photo library. Your device's restrictions do not allow access.")
+        @unknown default:
+            preconditionFailure("The app does not handle this new case provided by Apple")
+        }
+        presentImagePickerController()
+        
+    }
+    
+    @IBAction func recordAudioTapped(_ sender: UIButton) {
+        
+        guard titleTextField.text != "" else { return }
```

Comment:
here is another good spot for an alert controller, just letting the user know that they need to enter text in the titleTextfield otherwise they wont be able to record audio.